Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -2239,6 +2273,42 @@ listScheduledBookings (personId, _, cityId) mbLimit mbOffset mbFromDay mbToDay m
             price = priceObject.amountInt,
             priceWithCurrency = mkPriceAPIEntity priceObject
           }
+    getTodayScheduledBookings :: UTCTime -> Id DMOC.MerchantOperatingCity -> [VehicleVariant] -> LatLong -> Vehicle -> TransporterConfig -> [ServiceTierType] -> [DTC.TripCategory] -> Integer -> Integer -> Integer -> Flow [Maybe DRB.Booking]
+    getTodayScheduledBookings now mocCityId vehicleVariants dLoc vehicle transporterConfig possibleServiceTierTypes tripCategories limit offset safelimit = do
+      let nextDay = addDays 1 (utctDay now)
+          nextDayStartTime = UTCTime nextDay 0
+          redisKeys = createRedisKeysForCombinations now mocCityId tripCategories vehicleVariants
+          redisKeyForHset = createRedisKeyForHset now mocCityId
+          startScore = calculateSortedSetScore $ addUTCTime 1800 now
+          endScore = calculateSortedSetScore $ addUTCTime (3600 * 2) now
+          startScore2 = calculateSortedSetScore $ addUTCTime ((3600 * 2) + 1) now
+          end = calculateSortedSetScore nextDayStartTime
+
+      res <- mapM (\key -> Redis.zRangeByScoreByCount key startScore endScore offset safelimit) redisKeys
+      res2 <- mapM (\key -> Redis.zRangeByScoreByCount key startScore2 end offset limit) redisKeys
+
+      let parsedRes = mapMaybe (parseMember . decodeUtf8) (concat (res ++ res2))
+          nearbyBookings = take (fromIntegral limit) $ filterNearbyBookings now dLoc vehicle.variant transporterConfig.avgSpeedOfVehicle parsedRes possibleServiceTierTypes
+
+      if not $ null nearbyBookings
+        then Redis.hmGet redisKeyForHset nearbyBookings
+        else pure []
+
+    getTommorowScheduledBookings :: UTCTime -> UTCTime -> Id DMOC.MerchantOperatingCity -> [VehicleVariant] -> LatLong -> Vehicle -> TransporterConfig -> [ServiceTierType] -> [DTC.TripCategory] -> Integer -> Integer -> Flow [Maybe DRB.Booking]
+    getTommorowScheduledBookings now dayStartTime mocCityId vehicleVariants dLoc vehicle transporterConfig possibleServiceTierTypes tripCategories limit offset = do
+      let redisKeys = createRedisKeysForCombinations dayStartTime mocCityId tripCategories vehicleVariants
+          redisKeyForHset = createRedisKeyForHset dayStartTime mocCityId
+          startScore = calculateSortedSetScore dayStartTime
+          endScore = calculateSortedSetScore $ addUTCTime (3600 * 24) dayStartTime
+
+      res <- mapM (\key -> Redis.zRangeByScoreByCount key startScore endScore offset limit) redisKeys
+
+      let parsedRes = mapMaybe (parseMember . decodeUtf8) (concat res)
+          nearbyBookings = filterNearbyBookings now dLoc vehicle.variant transporterConfig.avgSpeedOfVehicle parsedRes possibleServiceTierTypes
+
+      if not $ null nearbyBookings
+        then Redis.hmGet redisKeyForHset nearbyBookings
+        else pure []
```

Comment:
Both these functions are very similar. Can you create one common function for these?